app/controllers/rooms_controller.rb

```diff
@@ -0,0 +1,69 @@
+class RoomsController < ApplicationController
+  before_action :set_room, only: [:edit, :update, :destroy, :archive]
+
+  def index
+    @rooms = Room.all
+  end
+
+  def show
+    @room = Room.find_by(slug: params[:slug])
+    raise ActiveRecord::RecordNotFound, "Couldn't find a room for that URL" unless @room
+    @topics = @room.topics.all
+  end
+
+  def new
+    @room = Room.new
+  end
+
+  def edit
+  end
+
+  def create
+    @room = Room.new(room_params)
+
+    respond_to do |format|
+      if @room.save
+        format.html { redirect_to @room, notice: 'Room was successfully created.' }
+        format.json { render :show, status: :created, location: @room }
+      else
+        format.html { render :new }
+        format.json { render json: @room.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def update
+    respond_to do |format|
+      if @room.update(room_params)
+        format.html { redirect_to root_url, notice: 'Room was successfully updated.' }
+        format.json { render :show, status: :ok, location: @room }
+      else
+        format.html { render :edit }
+        format.json { render json: @room.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def destroy
+    @room.destroy
+    respond_to do |format|
+      format.html { redirect_to root_url, notice: 'Room was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
+
+  def archive
+    topics = @room.topics.order('created_at DESC')
+    topics.where!('title ilike ?', "%#{params[:q]}%") if params[:q]
+    @grouped_topics = topics.group_by{ |t| t.created_at.to_date }
```

Comment:
Since renaming it to `archives` was such an easy change, I went ahead and did it already. Try it out, and let me know if you still think it'll be confusing for future devs and not worth the (probably negligible) user nicety of a cleaner URL.